Source/WebCore/inspector/agents/InspectorCSSAgent.cpp

```diff
@@ -432,10 +441,16 @@ bool InspectorCSSAgent::forcePseudoState(const Element& element, CSSSelector::Ps
     switch (pseudoClassType) {
     case CSSSelector::PseudoClassActive:
         return forcedPseudoState & PseudoClassActive;
-    case CSSSelector::PseudoClassFocus:
-        return forcedPseudoState & PseudoClassFocus;
     case CSSSelector::PseudoClassHover:
         return forcedPseudoState & PseudoClassHover;
+    case CSSSelector::PseudoClassFocus:
+        return forcedPseudoState & PseudoClassFocus;
+    case CSSSelector::PseudoClassFocusVisible:
+        return forcedPseudoState & PseudoClassFocusVisible;
+    case CSSSelector::PseudoClassFocusWithin:
+        return forcedPseudoState & PseudoClassFocusWithin;
+    case CSSSelector::PseudoClassTarget:
+        return forcedPseudoState & PseudoClassTarget;
```

Comment:
I know this is all existing logic, but it seems odd we have our own duplicate enum and this switch-case to map between the two. Could we instead store `CSSSelector::PseudoClass*` values directly into `m_nodeIdToForcedPseudoState`, and then here we could `return forcedPseudoState & pseudoClassType` - no switch statement necessary?

Reply:
sadly `CSSSelector::PseudoClass` is not able to be used as a bitmask, so we'd probably have to have something like `HashSet<CSSSelector::PseudoClassType>` which not great, but i suppose it's not that expensive to store (and this is likely not gonna be used more than a handful of times)